sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -160,25 +160,58 @@ export class EventHubConsumer {
   receive(onMessage: OnMessage, onError: OnError, abortSignal?: AbortSignalLike): ReceiveHandler {
     this._throwIfReceiverOrConnectionClosed();
     this._throwIfAlreadyReceiving();
+
+    // _throwIfReceiverOrConnectionClosed _should_ cover this case already, but throw an error if something unexpected happened.
+    if (!this._baseConsumer) {
+      const errorMessage =
+        `The EventHubConsumer for "${this._context.config.entityPath}" can no longer be used. ` +
+        `Please create a new EventHubConsumer using the "createConsumer" function on the EventHubClient.`;
+      throw new Error(errorMessage);
```

Comment:
Updated, though the main reason I had this check was so that the TypeScript compiler wouldn't complain that `this._baseConsumer` might be null. I've updated all of the `this._baseConsumer` accesses to include the non-null assertion (!) which makes me slightly sad :-(

Reply:
Then, we need a better solution than allowing baseConsumer to be `void 0`

Reply:
I now store the `_baseConsumer` in a local variable so I only needed to do the non-null assertion once. I still want to set it to undefined when `close` is called to help it get cleaned up by the GC.